src/components/environment-badge.tsx

```diff
@@ -37,3 +39,32 @@ export function EnvironmentBadge( { type, selected }: EnvironmentBadgeProps ) {
 
 	return <Badge className={ cx( getClassName() ) }>{ labels[ type ] }</Badge>;
 }
+
+export const useSiteEnvDetails = (
+	connectedSite: SyncSite
+): { AssignedEnvironmentBadge: () => JSX.Element; envType: EnvironmentType } => {
+	const { __ } = useI18n();
+
+	const envTypeValues = {
+		production: {
+			AssignedEnvironmentBadge: () => <EnvironmentBadge type="production" />,
+			envType: 'production',
+		},
+		staging: {
+			AssignedEnvironmentBadge: () => <EnvironmentBadge type="staging" />,
+			envType: 'staging',
+		},
+		sandbox: {
+			AssignedEnvironmentBadge: () => <EnvironmentBadge type="sandbox" />,
+			envType: 'sandbox',
+		},
+	} as const;
+
+	if ( connectedSite.isPressable ) {
+		return (
+			envTypeValues[ connectedSite.environmentType as EnvironmentType ] ?? envTypeValues.production
```

Comment:
Could we adjust the type on `SyncSite` so we don't have to do this cast?

Reply:
The problem is - `environmentType` doesn't exists for dotcom sites. Theoretically we can try to improve typing to distinguish dotcom and pressable sites, but it will take more effort, so I would keep it as is, sicne it's out of project. We shoudl focus only on project progress ATM.

Reply:
I think it's also not nice approach, since we are mixing types, since `environmentType` is null only for dotcom btu oposite for Pressable. So I think that if we want to improve it, then we shoudl do it in correct way with:

```
{
environmentType: 'production' | 'staging' | 'sandbox'
isPressable: true,
} |
{
isPressable: false,
}
```

But it will take muuch more effort to test teh whole coderbase and make fixes. Let's focus on the project.